src/viper/common.ml

```diff
@@ -8,3 +8,14 @@ let rec map_last ~f = function
   | [] -> []
   | [ x ] -> [ f x ]
   | x :: xs -> x :: map_last ~f xs
+
+let tup_con n =
+  Printf.sprintf "Tup$%d" n
+
+let tup_prj n i =
+  Printf.sprintf "tup$%d$%d" n i
```

Comment:
```suggestion
let tup_con_name n =
  Printf.sprintf "Tup$%d" n

let tup_prj_name n i =
  Printf.sprintf "tup$%d$%d" n i
```

Maybe it makes purpose of this helpers a bit clearer, because in `trans.ml` function `tup_prj` looks like subroutine that translate tuple projection.

Reply:
Done